### ISSUE TITLE
Surface configs not working

### ISSUE THREAD

**Issue by whichonespink44** [Team-RTG/Realistic-Terrain-Generation#1046]
```            #  [default: ]
            S:"RTG Surface: Filler Block"=minecraft:dirt

            #  [default: ]
            S:"RTG Surface: Filler Block Meta"=0

            #  [default: ]
            S:"RTG Surface: Top Block"=minecraft:grass

            #  [default: ]
            S:"RTG Surface: Top Block Meta"=0
```

![](https://cdn.discordapp.com/attachments/168326116761665536/247146694330810378/2016-11-13_10.12.53.png)

**Comment by srs-bsns** [comment]
In the case of vanilla Desert, the vanilla settings for the biome are passed to the surface initialiser, and there is no check for config settings. I'm assuming that at the time you felt in unnecessary under normal circumstances for Desert to have both the surface and fill blocks set to something other than minecraft:sand, which makes sense.

https://github.com/Team-RTG/Realistic-Terrain-Generation/blob/1.10.2-dev/src/main/java/rtg/world/biome/realistic/vanilla/RealisticBiomeVanillaDesert.java#L68

However, other biomes, such as ExtremeHills, do check for the config settings: https://github.com/Team-RTG/Realistic-Terrain-Generation/blob/1.10.2-dev/src/main/java/rtg/world/biome/realistic/vanilla/RealisticBiomeVanillaExtremeHills.java#L86

So the solution here is quite simple. Change the Desert (and any other affected biomes) to check, and set, it's top/filler blocks from the configs.

**Comment by whichonespink44** [comment]
Nice one - I thought that might be the case. There are probably a bunch of others like this that slipped through the cracks when we originally did the biome configs, so I'll go through them all and hook them up. I'm actually working on the 'RB Configs' stuff now, so I'll do this as part of that.

**Comment by whichonespink44** [comment]
This will be fixed when #1048 is merged.

**Comment by whichonespink44** [comment]
Fixed in 7d27130
